src/typingsInstaller/nodeTypingsInstaller.ts

```diff
@@ -84,6 +84,7 @@ namespace ts.server.typingsInstaller {
         private delayedInitializationError: InitializationFailedResponse | undefined;
 
         constructor(globalTypingsCacheLocation: string, typingSafeListLocation: string, typesMapLocation: string, npmLocation: string | undefined, validateDefaultNpmLocation: boolean, throttleLimit: number, log: Log) {
+            // TODO: This just uses sys, I'm not sure how to circumvent that

```

Comment:
This is correct implementation of typing installer that installs the packages using npm install on the disk

I think when you have fsHost = vfs 
I think we need typing installer that just converts the typing installer into some kind of event that vscode/vs listen and do the work and give us files and their text to load in the vfs